pages/starsearch.tsx

```diff
@@ -129,25 +206,35 @@ function Chatbox({ author, content, userId }: { author: "You" | "StarSearch"; co
   return (
     <li className="flex gap-2 items-start my-4">
       {renderAvatar()}
-      <Card className="flex flex-col grow bg-white z-10 p-2 lg:p-4">
+      <Card className="flex flex-col grow bg-white z-10 p-2 lg:p-4 lg:max-w-5xl">
         <h3 className="font-semibold text-sauced-orange">{author}</h3>
         <p>{content}</p>
       </Card>
     </li>
   );
 }
 
-function StarSearchInput({ onSubmit }: { onSubmit: () => void }) {
+function StarSearchInput({
+  isRunning,
+  onSubmitPrompt,
+}: {
+  isRunning: boolean;
+  onSubmitPrompt: (prompt: string) => void;
+}) {
+  const [promptInput, setPromptInput] = useState<string>("");
   return (
     <section className="w-full h-fit max-w-4xl px-1 py-[3px] rounded-xl bg-gradient-to-r from-sauced-orange via-amber-400 to-sauced-orange">
       <div className="w-full h-fit flex justify-between rounded-lg">
         <input
+          required
```

Comment:
This will work and require less handling of events, i.e. no `onChange` or `onKeyDown` on the input and no need to store the input's value in component state.

instead of

```jsx
      <div className="w-full h-fit flex justify-between rounded-lg">
        <input
          required
          type="text"
          disabled={isRunning}
          placeholder="Ask a question"
          onChange={(e) => setPromptInput(e.target.value)}
          onKeyDown={(e) => e.key === "Enter" && onSubmitPrompt(promptInput)}
          className="p-4 border focus:outline-none grow rounded-l-lg border-none"
        />
        <button disabled={isRunning} onClick={() => onSubmitPrompt(promptInput)} className="bg-white p-2 rounded-r-lg">
          <MdOutlineSubdirectoryArrowRight className="rounded-lg w-10 h-10 p-2 bg-light-orange-3 text-light-orange-10" />
        </button>
      </div>
```

use a form and move a few things around.

```jsx
      <form
        onSubmit={(event) => {
          event.preventDefault();
          const form = event.currentTarget;
          const formData = new FormData(form);
          onSubmitPrompt(formData.get("prompt") as string);
          form.reset();
        }}
        className="w-full h-fit flex justify-between rounded-lg"
      >
        <input
          required
          name="prompt"
          type="text"
          disabled={isRunning}
          placeholder="Ask a question"
          className="p-4 border focus:outline-none grow rounded-l-lg border-none"
        />
        <button disabled={isRunning} className="bg-white p-2 rounded-r-lg">
          <MdOutlineSubdirectoryArrowRight className="rounded-lg w-10 h-10 p-2 bg-light-orange-3 text-light-orange-10" />
        </button>
      </form>
```

TLDR, we should leverage forms all the time for any kind of submission even if we aren't currently doing a full page reload.